wasm/src/algorithms/bhp/mod.rs

```diff
@@ -125,4 +123,168 @@ mod tests {
             assert_eq!(commit_to_group_1024, Group::from(native_commit_to_group_1024));
         }
     }
+
+    #[wasm_bindgen_test]
+    fn test_wasm_literal_bhp_hashes_equal_native_hashes() {
+        // Create a field element and a scalar element.
+        let scalar = Scalar::from_string(SCALAR_FIELD_ELEMENT).unwrap();
+
+        // Create all exported BHP hasher instances.
+        let bhp256 = BHP256::new();
+        let bhp512 = BHP512::new();
+        let bhp768 = BHP768::new();
+        let bhp1024 = BHP1024::new();
+
+        // Create all native BHP hasher instances.
+        let native_bhp256 = BHP256Native::setup("AleoBHP256").unwrap();
+        let native_bhp512 = BHP512Native::setup("AleoBHP512").unwrap();
+        let native_bhp768 = BHP768Native::setup("AleoBHP768").unwrap();
+        let native_bhp1024 = BHP1024Native::setup("AleoBHP1024").unwrap();
+
+        let address = Address::zero();
+
+        let literals = vec![
+            Literal::Address(address),
+            Literal::Boolean(Boolean::new(false)),
+            Literal::Field(snarkvm_console::types::Field::from_u8(2)),
+            Literal::Group(
+                snarkvm_console::types::Group::from_str(
+                    "7243206743250892049702172909169115544952822465955921992746259936160368017976group",
+                )
+                .unwrap(),
+            ),
+            Literal::Scalar(snarkvm_console::types::Scalar::from_field_lossy(&snarkvm_console::types::Field::from_u8(
+                2,
+            ))),
+        ];
+
+        let native_literals = vec![
+            LiteralNative::Address(address),
+            LiteralNative::Boolean(Boolean::new(false)),
+            LiteralNative::Field(FieldNative::from_u8(2)),
+            LiteralNative::Group(
+                GroupNative::from_str(
+                    "7243206743250892049702172909169115544952822465955921992746259936160368017976group",
+                )
+                .unwrap(),
+            ),
+            LiteralNative::Scalar(ScalarNative::from_field_lossy(&snarkvm_console::types::Field::from_u8(2))),
+        ];
+
+        for i in 0..literals.len() {
+            let literal_bits = Plaintext::Literal(literals[i].clone(), OnceCell::new()).to_bits_le();
```

Comment:
Same comment here as in the Poseidon hash comments, this is just the Plaintext type directly from SnarkVM. We want to test the wasm representation, so the definition above of the `literal` variable should create our exported `Plaintext` structs instead with the `to_plaintext` method.